src/store/undoRedo/index.ts

```diff
@@ -38,6 +38,12 @@ function getChangedModulesKeys(snapshotA: ISnapshot, snapshotB: ISnapshot): stri
   return getChangedKeys(modulesA, modulesB)
 }
 
+function shouldAlwaysTriggerSnapshot(changedKeys: string[]): boolean {
+  return changedKeys.some((key) => (
+    /^flows.flows.\d+.blocks.\d+.exits.\d+.destination_block$/.test(key)
+  ))
+}
+
```

Comment:
@bulatgab creating and deleting block connections right away fell under the "we changed the same thing" category; I created a list of overrides with store paths that should always produce a new snapshot.